src/common/wrap/wrap-promise.js

```diff
@@ -69,9 +69,9 @@ export function wrapPromise (sharedEE) {
       const prevStaticFn = prevPromiseObj[method]
       WrappedPromise[method] = function (subPromises) { // use our own wrapped version of "Promise.all" and ".race" static fns
         let finalized = false
-        subPromises?.forEach(sub => {
-          // eslint-disable-next-line
-          this.resolve(sub).then(setNrId(method === 'all'), setNrId(false));
+
+        Array.from(subPromises || []).forEach(sub => {
```

Comment:
Not critical, but why not use `for...of` that works with any `iterable` class?

Reply:
`for..of` was tried and yea it works but its also a .21% gzip code size increase for the spa loader due to the need for additional error prevention logic.